Add new attribute: inhibit_automatic_where_clause

vi opened the pull request:
Allows to opt out of where clause, in cases where it gets in the way.

My use case:

```rust

#[delegatable_trait]
trait SomeTrait{}

#[derive(ambassador::Delegate)]
#[delegate(SomeTrait,inhibit_automatic_where_clause="true")]
pub struct DynSomeTrait(pub Box<dyn SomeTrait + Send>);
```

It allows user to request structural typing (like with `target="self"`), but with other targets instead of `self`.

If this feature is welcome in `ambassador`, I can also provide in this pull request:

* Similar attribute for impl delegation
* Documentation
* Test

vi committed:
Add new attribute: inhibit_automatic_where_clause

dewert99 commented on ambassador/src/delegate_shared.rs:
```diff
@@ -55,6 +56,9 @@ impl<T: DelegateTarget> DelegateArgs<T> {
                             res.generics.extend(generics_val);
                         }
+                        "inhibit_automatic_where_clause" => {
+                            res.inhibit_automatic_where_clause = true;
```
Does this check to make the value was "true" (`inhibit_automatic_where_clause="true"`).  I don't think it should be enabled if someone has `inhibit_automatic_where_clause="false"`)

dewert99 replied:
Other than that, if you want to add documentation and tests, I'd be happy to merge this.

vi committed:
Add a test for inhibit_automatic_where_clause

vi committed:
Additional test to ensure alternative way of doing dyn delegation works.

vi committed:
Properly parse inhibit_automatic_where_clause value.

vi committed:
Remove possibility of double negation from API

vi committed:
Document `automatic_where_clause`.

vi commented:
@dewert99, Done.

Some changes:

* I changed `inhibit_automatic_where_clause="true"` to `automatic_where_clause="false"` to avoid double negation in API. Internal name stays `inhibit_automatic_where_clause` due to `Default` implementation.
* I have added a test ensures that a workaround of attaining dyn delegation this without `automatic_where_clause="false"` works and will not be accidentally broken (i.e. by implementing automatic where clauses for `#[delegate_to_methods]`) without a major semver increase.

dewert99 commented:
This looks great do you think you could run `rustfmt` and then I can pull this

vi committed:
cargo fmt
